agent/engine/docker_task_engine.go

```diff
@@ -2946,3 +2954,66 @@ func (engine *DockerTaskEngine) getDockerID(task *apitask.Task, container *apico
 	}
 	return dockerContainer.DockerID, nil
 }
+
+// Sets CloudWatch Logs dual stack endpoint as awslogs-endpoint option in logging config.
+// This is needed because awslogs driver that we consume from Docker does not support
+// dual stack endpoints.
+func setAWSLogsDualStackEndpoint(
+	task *apitask.Task, container *apicontainer.Container, hostConfig *dockercontainer.HostConfig,
+) {
+	if hostConfig.LogConfig.Config[awsLogsEndpointKey] != "" {
```

Comment:
The behavior is little different from how agent handles "awslogs-endpoint" assignment in isolated regions.

Here we have this additional check if the property exists in LogConfig. However for isolated regions we assign regardless, i.e. Agent has the control over this parameter.

Should both assignments follow similar pattern and order?

Reply:
Customers are able to set `awslogs-endpoint` in their task definitions. I added this check so that their configured endpoint is not overridden by Agent.

Reply:
IMO this check should exist for isolated regions case too.

Reply:
@xxx0624 It's not documented but it works

Reply:
The documentation says "some of the options" (emphasis mine) - 

> *Some* of the options you can specify when you use the awslogs log driver to route logs to Amazon CloudWatch include the following:

https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_LogConfiguration.html

Reply:
TIL